mind_matter_api/models/users.py

```
@@ -6,16 +6,32 @@ class User(db.Model):
 
     # Example relationships:
-    consent = db.relationship("UserConsent", back_populates="user", uselist=False)
     survey_responses = db.relationship("SurveyResponse", back_populates="user")
     mood_logs = db.relationship("MoodActivityLog", back_populates="user")
     notifications = db.relationship("Notification", back_populates="user")
     campaign_participation = db.relationship("CampaignParticipation", back_populates="user")
     incentives = db.relationship("Incentive", back_populates="user")
     reward_redemptions = db.relationship("RewardRedemption", back_populates="user")
+    
+    def set_password(self, password: str):
+        self.password = generate_password_hash(password)
```

Ruff (0.8.2)

Undefined name `generate_password_hash`

(F821)